crates/solver/src/arguments.rs

```diff
@@ -299,6 +299,26 @@ pub struct Arguments {
     /// value 0.0012 ETH and 0.0016 ETH equivalent.
     #[clap(long, env, default_value = "0")]
     pub solution_comparison_decimal_cutoff: u16,
+
+    /// The s3_instance_upload_* arguments configure how quasimodo instances should be uploaded to
+    /// AWS S3. They must either all be set or all not set. If they are set then every instance sent to
+    /// Quasimodo as part of auction solving is also uploaded to S3.
```

Comment:
Not sure why we tie the s3 upload to quasimodo.
This could also be used to debug the naive or baseline solver. And overall I think keeping that data is valuable anyway. So unless the s3 storage is very expensive I would err on the side of caution and store all of the auctions.

Reply:
The instance is meaningful for http solvers. And between those they all share the same instance. It's not useful to upload it multiple times. The reason I'm tying it to Quasimodo is that that is the main reason we keep the instances in the first place as we do now.

Reply:
It would make a difference if we disabled Quasimodo but uploading a generic instance without tying it to a solver would be a bigger refactor.